pkg/services/ssosettings/ssosettingsimpl/service.go

```diff
@@ -84,6 +84,33 @@ func (s *SSOSettingsService) GetForProvider(ctx context.Context, provider string
 	return storeSettings, nil
 }
 
+func (s *SSOSettingsService) GetForProviderWithRedactedSecrets(ctx context.Context, provider string) (*models.SSOSettings, error) {
+	storeSettings, err := s.store.Get(ctx, provider)
+
+	if errors.Is(err, ssosettings.ErrNotFound) {
+		settings, err := s.loadSettingsUsingFallbackStrategy(ctx, provider)
+		if err != nil {
+			return nil, err
+		}
+
+		return settings, nil
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	storeSettings.Source = models.DB
```

Comment:
You can call `GetForProvider` instead of copy-pasting its code.
```suggestion
	storeSettings, err := s.GetForProvider(ctx, provider)
	if err != nil {
		return nil, err
	}
```